ludwig/schema/model_config.py

```diff
@@ -172,17 +172,6 @@ def __init__(self, config_dict: ModelConfigDict):
                 ):
                     raise ValidationError("GBM Model trainer must be of type: 'lightgbm_trainer'")
 
-                for feature in self.input_features.to_dict().keys():
```

Comment:
We should have the feature encoder tell us via its interface what model types it accepts.

Reply:
@tgaddair I just saw your latest comments on my PR and this is what I was thinking as well. As you said, it'll be a relatively large refactor. CC @connor-mccorm